tests/e2e/test_actions.py

```diff
@@ -384,3 +384,36 @@ def test_shutdown_action_now(update_environment):
         assert (
             "This condition should not fire" not in result.stdout
         ), "a post-shutdown condition fired when it should not have"
+
+
+@pytest.mark.e2e
+def test_inventory_as_dir():
+    """
+    Execute a rulebook that contains a run_playbook action with an inventory
+    directory instead of a file.
+    """
+
+    rulebook = utils.BASE_DATA_PATH / "rulebooks/test_inventory_as_dir.yml"
+    inventory = utils.BASE_DATA_PATH / "inventories/inventory_as_dir"
+    cmd = utils.Command(
+        rulebook=rulebook,
+        inventory=inventory,
+    )
+
+    LOGGER.info(f"Running command: {cmd}")
+    result = subprocess.run(
+        cmd,
+        timeout=DEFAULT_CMD_TIMEOUT,
+        capture_output=True,
+        cwd=utils.BASE_DATA_PATH,
+        text=True,
+    )
+
+    assert result.returncode == 0
+    assert not result.stderr
+    assert (
+        "hostvar_value" in result.stdout
+    ), "hostvar_value not found in stdout"
+    assert (
+        "groupvar_value" in result.stdout
+    ), "groupvar_value not found in stdout"
```

Comment:
We should use the `with check:` context manager for these assertions.

Reply:
I don't think is necessary in this case @ttuffin We use "with check" when we want to reuse the same rulebook for different testcases for efficiency purposes. This case is a single testcase and all the assertions belongs to the same testcase.

Reply:
No we use `with check` whenever we check for the expected output so that the whole test isn't failed if one of the expected outputs is false. Have a look at our other e2e test cases. We use it in those and we should do the same here.

Reply:
Hi @ttuffin I think it is misunderstanding about the purpose of the plugin `pytest-check`

> so that the whole test isn't failed if one of the expected outputs is false.
That is why when we reuse a rulebook to perform different testcases in the same test we need to not fail of one of them and continue with the test. The regular way is to have each test case in a different test, but we wanted to put together different testcases in the same rulebook for efficiency reasons (for example one rulebook to test all the actions) and is when we used that plugin. That plugin is only intended for all those places where we want to run different testcases reusing the test/rulebook. 

This is not the case, this is a single specific test case and it is a regular test, without the need of that plugin. If other tests use it without need it, it is a mistake there.